src/catalog/src/durable/persist.rs

```diff
@@ -777,6 +792,7 @@ impl UnopenedPersistCatalogState {
         persist_client: PersistClient,
         organization_id: Uuid,
         version: semver::Version,
+        deploy_generation: Option<u64>,
```

Comment:
This one line is the reason the LoC is so high on this PR. We used to pass in the deploy generation when opening the durable catalog. Now we pass it in even earlier, before opening the catalog when we initialize it, which is handled by this function. This changed the API of a handful of heavily used methods.

The reason is so that we can fence the catalog by deploy version, before we even open the catalog. Maybe that was unnecessary and we should have kept the API as it was. This feels a bit safer for some reason.